pkg/plugin/testfiles/20-setRouteHeader.yaml

```diff
@@ -0,0 +1,89 @@
+rollout:
+  apiVersion: argoproj.io/v1alpha1
+  kind: Rollout
+  metadata:
+    name: demo
+    namespace: gloo-mesh
+  spec:
+    replicas: 3
+    selector:
+      matchLabels:
+        app: demo
+    template:
+      metadata:
+        labels:
+          app: demo
+      spec:
+        containers:
+        - image:  kodacd/argo-rollouts-demo-api:v1
+          imagePullPolicy: IfNotPresent
+          name: demo
+          ports:
+          - containerPort: 8080
+    strategy:
+      canary:
+        canaryService: canary
+        stableService: stable
+        trafficRouting:
+          plugins:
+            solo-io/glooplatform:
+              canaryDestination:
+                ref:
+                  name: canary
+                  namespace: gloo-demo-rollout
+                port:
+                  number: 8099
+              routeTableSelector:
+                name: demo
+                namespace: gloo-mesh
+        steps:
+        - setWeight: 10
+        - setHeaderRoute:
+            name: "set-header-canary" # we should document that if this route already exists in their routeTable we may have a problem. I think we need to take the route specified in the routeTableSelector, copy it, adjust the name to this name, add a header match to the existing matches, ensure the dst in canary... 
+            match:
+            - headerName: version
+              headerValue:
+                exact: canary
+        - pause: {}
+        - setWeight: 100
+
+routeTable:
+  apiVersion: networking.gloo.solo.io/v2
+  kind: RouteTable
+  metadata:
+    name: default
+    namespace: gloo-mesh
+  spec:
+    http:
+    - name: demo
+      matchers:
+        - uri:
+            prefix: /demo
+      labels:
+        route: demo
+      forwardTo:
+        pathRewrite: /
+        destinations:
+        - ref:
+            name: stable
+            namespace: gloo-rollout-demo
+          port:
+            number: 8080
+          kind: SERVICE
+stepAssertions:
+# - step: 1
+#   assert:
+#   - path: $.spec.http[0].forwardTo.destinations
+#     exp: len == 2
+#   - path: $.spec.http[0].forwardTo.destinations[?(@.ref.name=="stable")].weight
+#     exp: value == 90
+#   - path: $.spec.http[0].forwardTo.destinations[?(@.ref.name=="canary")].weight
+#     exp: value == 10
```

Comment:
Are these meant to be commented out?

Reply:
ah, yeah. these are asserted in other tests. I'll remove though